packages/nimble-components/src/select/index.ts

```diff
@@ -571,35 +590,49 @@ export class Select
                 if (!this.open) {
                     break;
                 }
+
+                this.toggleNewActiveOption(
+                    this.indexWhenOpened ?? this.selectedIndex
```

Comment:
I find the distinction between `this.indexWhenOpen` and `this.selectedIndex` a bit confusing now. Wouldn't those always be the same value since `this.selectedIndex` doesn't change while keying through the options?

Reply:
Good call! I've removed the `indexWhenOpened` field now. Hopefully, the result feels more understandable.